openml/flows/sklearn_converter.py

```diff
@@ -506,14 +507,35 @@ def _extract_information_from_model(model):
     for k, v in sorted(model_parameters.items(), key=lambda t: t[0]):
         rval = sklearn_to_flow(v, model)
 
-        if (isinstance(rval, (list, tuple))
+        def flatten_all(list_):
+            """ Flattens arbitrary depth lists of lists (e.g. [[1,2],[3,[1]]] -> [1,2,3,1]). """
+            for el in list_:
+                if isinstance(el, (list, tuple)):
+                    yield from flatten_all(el)
+                else:
+                    yield el
+
+        # In case rval is a list of lists (or tuples), we need to identify two situations:
+        # - sklearn pipeline steps, feature union or base classifiers in voting classifier.
+        #   They look like e.g. [("imputer", Imputer()), ("classifier", SVC())]
+        # - a list of lists with simple types (e.g. int or str), such as for an OrdinalEncoder
+        #   where all possible values for each feature are described: [[0,1,2], [1,2,5]]
+        is_non_empty_list_of_lists_with_same_type = (
+            isinstance(rval, (list, tuple))
             and len(rval) > 0
             and isinstance(rval[0], (list, tuple))
-            and all([isinstance(rval[i], type(rval[0]))
-                     for i in range(len(rval))])):
+            and all([isinstance(rval_i, type(rval[0])) for rval_i in rval])
+        )
 
-            # Steps in a pipeline or feature union, or base classifiers in
-            # voting classifier
+        nested_list_of_simple_types = (
```

Comment:
Okay, so this iteratively checks if all elements in the list are simple types? If yes, could you maybe add a comment here that it does so?

Reply:
You are correct, I'll add the requested comment.